Emby.Naming/Video/CleanStringParser.cs

```diff
@@ -25,26 +25,54 @@ public static bool TryClean([NotNullWhen(true)] string? name, IReadOnlyList<Rege
                 return false;
             }
 
-            var len = expressions.Count;
-            for (int i = 0; i < len; i++)
+            // Iteratively remove extra cruft until we're left with the string
+            // we want.
+            newName = ReadOnlySpan<char>.Empty;
+            const int maxTries = 100;       // This is just a precautionary
+                                            // measure. Should not be neccesary.
+            var loopCounter = 0;
+            for (; loopCounter < maxTries; loopCounter++)
```

Comment:
Why a loop? Regex.Replace will remove all occurrences. Isn't that essentially what you're doing?

Reply:
A loop makes the regular expressions less reliant on the order they're matched, which I thought would be easier for maintenance. For example, I  only want to match the episode marker "E01-E016" at the "end" of the filename, but "end" here could be before a "[720p]" tag (e.g., "Show E01-E16 [720p]"). Without the loop, this requires that the "E01-E16" regex come strictly after the "[...]" regex.

If that's not an issue, I can change the loop, and just add a comment saying that the order of the regexs matter.

Reply:
Don't really need to a comment explaining that the order of the regexes matter, the same is true for the episode parser regexes.

Reply:
Got rid of loop in newest edition